include/arch/arm/arch/64/mode/machine/registerset.h

```diff
@@ -232,6 +233,33 @@ extern const register_t msgRegisters[];
 extern const register_t frameRegisters[];
 extern const register_t gpRegisters[];
 
+#ifdef ARM_BASE_CP14_SAVE_AND_RESTORE
+typedef struct debug_register_pair {
+    word_t cr, vr;
+} debug_register_pair_t;
+
+// @alwin: Copied from armv7, necessary?
```

Comment:
This was introduced by commit f99f6790abe9c0d8b27a0aee0b6b29a0dc73c211 for 32-bit code. I would remove it, unless the variation of the value of `seL4_NumExclusiveWatchpoints` would change the TCB size. That is, the `seL4_TCBBits` define should always be correct, no matter which CPU variant is used.

Reply:
Ah, I think I understand now. Since we bump the TCB size up as a result of ARM_BASE_CP14_SAVE_AND_RESTORE, we need to make sure the actual size of the TCB matches what we say. 

I think the ARMv8 manual says that all CPUs that implement the spec need to have a minimum of 4 hardware breakpoint and watchpoint registers, so I just need to check if with this many, the TCB size needs to be bumped up, and if so, I can get rid of the padding.